coding_style/README.adoc

```diff
@@ -161,6 +166,17 @@ Rationale:: https://yaml.org/type/bool.html[YAML 1.1] allows all variants wherea
   your task will likely need options such as `changed_when:` and maybe `check_mode:`).
 * Anytime `command` or `shell` modules are used, add a comment in the code with justification to help with future maintenance.
 * Use the `| bool` filter when using bare variables (expressions consisting of just one variable reference without any operator) in `when`.
+* No single task file should contain more than 30 tasks.  Complex task files should be broken down into discrete parts.
+* The maximum block depth within a single task file should be 2
+* Bracket notation is preferred over dot notation for value retrieval (e.g. `item[‘key’]` vs. `item.key`)
++
+[%collapsible]
+====
+Rationale::
+Dot notation will fail in some cases (such as when a variable name includes a hyphen) and it's better to stay consistent than to switch between the two options within a role or playbook.
+For an example, see https://blog.networktocode.com/post/Exploring-Jinja-Variable-Syntax-in-Ansible
```

Comment:
```suggestion
Example:: see https://blog.networktocode.com/post/Exploring-Jinja-Variable-Syntax-in-Ansible
```

Reply:
I personally prefer concrete numbers. You can choose to change/ignore them, but it makes it easier to set an actual numerical limit rather than a "this file feels a bit too long" type of argument.

Maybe put a limit on number of lines in a file instead of/in addition to number of tasks? They can range from 2 lines (name, ansible.builtin.command: foo) to very long invocations with lots of parameters and so on.